lib/test.rb

```diff
@@ -882,7 +882,7 @@ def cleanup_after
       @category = __method__
       return if @skip_cleanup_after
 
-      if ENV["HOMEBREW_AZURE_PIPELINES"] || ENV["HOMEBREW_GITHUB_ACTIONS"]
+      if ENV["HOMEBREW_GITHUB_ACTIONS"] && !ENV["HOMEBREW_SELF_HOSTED_RUNNER"]
```

Comment:
```suggestion
      if ENV["HOMEBREW_GITHUB_ACTIONS"] && !ENV["GITHUB_ACTIONS_SELF_HOSTED_RUNNER"]
```

Reply:
This is a runner that Homebrew administers, not Github, so I don't think it makes sense to call it a "github" variable.

Reply:
All `HOMEBREW_` variables show up in `brew config` unless specifically blacklisted. It's not configuration for Homebrew itself, really, and we don't need the `HOMEBREW_` prefix because `test-bot` doesn't have environment filtering. Could do `GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED_RUNNER` or something?